app/src/main/java/id/rizmaulana/covid19/ui/overview/DashboardViewModel.kt

```diff
@@ -27,47 +33,62 @@ class DashboardViewModel(
-        get() = _dailyListData
+    private val _liveItems = MutableLiveData<List<BaseViewItem>>()
+    val items: LiveData<List<BaseViewItem>>
+        get() = _liveItems
 
-    private val _overviewData = MutableLiveData<CovidOverview>()
-    val overviewData: LiveData<CovidOverview>
-        get() = _overviewData
+    fun loadOverviewAndDaily() {
+        val overviewObservable = appRepository.overview()
+            .subscribeOn(schedulerProvider.io())
+            .observeOn(schedulerProvider.io()) //all stream below will be manage on io thread
+            .map { mapOverviewResponse(it) }
 
-    fun getOverview() {
-        appRepository.overview()
-            .observeOn(schedulerProvider.ui())
-            .doOnSubscribe {
-                val cache = appRepository.getCacheOverview()
-                if (cache == null) _loading.postValue(true) else {
-                    _overviewData.postValue(cache)
-                }
-            }
-            .doFinally { _loading.postValue(false) }
-            .subscribe({
-                _overviewData.postValue(it)
-            }, {
-                _errorMessage.postValue(Constant.ERROR_MESSAGE)
-            })
-            .addTo(compositeDisposable)
-    }
+        val dailyObservable = appRepository.daily()
+            .subscribeOn(schedulerProvider.io())
+            .observeOn(schedulerProvider.io()) //all stream below will be manage on io thread
+            .map { mapDailyResponse(it) }
+
+        Observable.combineLatest(overviewObservable, dailyObservable, BiFunction<OverviewItem?, List<BaseViewItem>?, List<BaseViewItem>> { overview, daily ->
+            val items: MutableList<BaseViewItem> = mutableListOf()
+
+            //GENERATE SCREEN POSITION HERE
+            items.add(overview)
+            items.add(TextItem("Daily Updates"))
+            items.addAll(daily)
 
-    fun getDailyUpdate() {
-        appRepository.daily()
-            .observeOn(schedulerProvider.ui())
-            .doOnSubscribe {
-                appRepository.getCacheDaily()?.let { data ->
-                    _dailyListData.postValue(data)
-                }
-            }
-            .subscribe({
-                _dailyListData.postValue(it)
-            }, {
-                _errorMessage.postValue(Constant.ERROR_MESSAGE)
-            })
-            .addTo(compositeDisposable)
+            return@BiFunction items.toList()
+        })
+        .observeOn(schedulerProvider.ui()) //go back to ui thread
+        .subscribe({
+            _liveItems.postValue(it)
+        }, {
+            _errorMessage.postValue(Constant.ERROR_MESSAGE)
+        }).addTo(compositeDisposable)
     }
 
+    private fun mapOverviewResponse(response: CovidOverview?) = OverviewItem(
+        response?.confirmed?.value ?: 0,
+        response?.recovered?.value ?: 0,
+        response?.deaths?.value ?: 0)
 
+    private fun mapDailyResponse(responses: List<CovidDaily>): List<BaseViewItem> {
```

Comment:
It also can be done this way

```
private fun mapDailyResponse(responses: List<CovidDaily>) = response.map { response ->
DailyItem(
                    response.objectid,
                    response.deltaConfirmed,
                    response.deltaRecovered,
                    response.mainlandChina,
                    response.otherLocations,
                    response.reportDate,
                    response.incrementRecovered,
                    response.incrementConfirmed
                )
}
```

Reply:
and I advise to move it with a new mapper class

Reply:
Is it something like this d4c412f ? @isfaaghyth @rizmaulana